iroh-embed/src/p2p.rs

```diff
@@ -59,6 +90,16 @@ impl P2pService {
         self.addr.clone()
     }
 
+    /// Returns a Receiver of NetworkEvents that have been emitted by this p2p service.
+    ///
+    /// This can be used to examine NetworkEvents that have occured
+    pub async fn network_events(&mut self) -> Receiver<NetworkEvent> {
+        let (s, r) = channel(512);
+        let mut events = self.events.lock().await;
+        events.push(s);
+        r
+    }
+
```

Comment:
@flub 
needed a way to expose the `NetworkEvents` `Receiver` to `iroh-share`.
This was one way I thought of doing it.

The other would be to add an rpc & api `network_events`method that would return a stream of `NetworkEvents`. 

After implementing the above, I'm leaning toward refactoring it to go the rpc/api route, but  was wondering if you had any opinions before I re-wrote it tomorrow.

Reply:
Channels are great, though the p2p `Node::run` (unsurprisingly) takes `&mut self` which means you have to get out the events channel out before invoking run which is rather meh.

Channels also mean you can only use the in-memory version of the p2p node.  That is of course the current idea of iroh-embed, though I think I agree using the irpc is nicer and more future proof (and internally they're just flume channels for iroh-embed so no real efficiency penalty).

So that's a verbose way of coming the the same conclusion, changing it to irpc would probably be nice :)